src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingManager.cs

```diff
@@ -89,8 +89,7 @@ internal class PersistentStreamPullingManager : SystemTarget, IPersistentStreamP
             // Remove cast once we cleanup
             queueAdapter = qAdapter.Value;
             await this.queueBalancer.Initialize(this.streamProviderName, this.adapterFactory.GetStreamQueueMapper(), config.SiloMaturityPeriod, this.providerConfig);
-            var meAsQueueBalanceListener = this.AsReference<IStreamQueueBalanceListener>();
-            queueBalancer.SubscribeToQueueDistributionChangeEvents(meAsQueueBalanceListener);
+            queueBalancer.SubscribeToQueueDistributionChangeEvents(this);
```

Comment:
The former of course.
And the reason is not that it is better, but to be consistent with all the rest that we do already.
For regular grain on the same silo: do wee call directly and ask the grain to ensure its own single threaded? WE call vei addressable context anywhere, always, and therefore we should simply be consistent and keep doing it.